modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/mapping/AssignmentsResolver.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine.exec.mapping;
+
+import org.jetbrains.annotations.Nullable;
+
+/** Calculate assignments according to supplied data. */
+interface AssignmentsResolver<RowT> {
+    /**
+     * Append object assignments to be calculated for.
+     *
+     * @param value The object  for which assignments will be calculated.
+     */
+    void append(@Nullable Object value);
+
+    /**
+     * Calculate assignment based on appending objects.
+     *
+     * @return Resolved assignment.
+     */
+    int getPartition();
```

Comment:
Is reset method is missing or it is not needed? Without it a caller should to find a way to create an instance of  AssignmentsResolver to calculate a partition for another set of values of colocation keys/

Reply:
Ignore this. Reset method is hidden inside `append`.